engine/e2e-test/test_cli_engine_list.py

```diff
@@ -9,6 +9,7 @@ class TestCliEngineList:
     @pytest.fixture(autouse=True)
     def setup_and_teardown(self):
         # Setup
+        stop_server()
```

Comment:
Hmm, I think we should make every test in isolation, which mean the test which start the server will have to stop the server itself.

Reply:
I agree. But some race condition in the tests make `stop_server` is not called. I cannot reproduce in the local machine, so make the e2e pass first